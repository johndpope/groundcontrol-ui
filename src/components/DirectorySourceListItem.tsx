// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import graphql from "babel-plugin-relay/macro";
import React from "react";
import { Button, List } from "semantic-ui-react";

import { createFragmentContainer } from "react-relay";

import { DirectorySourceListItem_item } from "./__generated__/DirectorySourceListItem_item.graphql";

export interface IProps {
  item: DirectorySourceListItem_item;
  onDelete: (values: IProps) => any;
}

export function DirectorySourceListItem(props: IProps) {
  const{ item: { directory}, onDelete } = props;
  const handleDelete = () => onDelete({ ...props });

  return (
    <List.Item>
      <List.Content>
        <Button
          floated="right"
          icon="delete"
          color="pink"
          size="small"
          onClick={handleDelete}
        />
        <List.Header>{directory}</List.Header>
        <List.Description>Directory</List.Description>
      </List.Content>
    </List.Item>
  );
}

export default createFragmentContainer(DirectorySourceListItem, graphql`
  fragment DirectorySourceListItem_item on DirectorySource {
    id
    directory
  }`,
);
