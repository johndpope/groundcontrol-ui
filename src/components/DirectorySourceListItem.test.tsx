// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { shallow } from "enzyme";
import React from "react";
import { mocked } from "ts-jest/utils";

import {  mockQueryPropAttrs, mockRelayProp } from "../testing/relay";

import { DirectorySourceListItem } from "./DirectorySourceListItem";

const props = {
  item: {
    ...mockQueryPropAttrs(),
    directory: "directory",
    id: "id",
  },
  onDelete: jest.fn(),
  relay: mockRelayProp(),
};

beforeEach(() => {
  mocked(props.onDelete).mockClear();
});

describe("<DirectorySourceListItem />", () => {

  it("renders correctly", () => {
    const wrapper = shallow(<DirectorySourceListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("triggers onDelete when the delete button is clicked", () => {
    const wrapper = shallow(<DirectorySourceListItem {...props} />);
    wrapper.find("Button[icon='delete']").simulate("click");
    expect(props.onDelete).toBeCalledTimes(1);
    expect(props.onDelete).toBeCalledWith(props);
  });

});
