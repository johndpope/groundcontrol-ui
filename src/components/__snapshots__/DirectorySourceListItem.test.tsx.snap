// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DirectorySourceListItem /> renders correctly 1`] = `
<ListItem>
  <ListContent>
    <Button
      as="button"
      color="pink"
      floated="right"
      icon="delete"
      onClick={[Function]}
      size="small"
    />
    <ListHeader>
      directory
    </ListHeader>
    <ListDescription>
      Directory
    </ListDescription>
  </ListContent>
</ListItem>
`;
